### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by BhasherBEL**
## CodeWe - version 2.0.1
**Front-End**
 - Multiple bugfix and prevent on editor

**Back-End**
 - http (:80) -> https (:443)
 - Some minor bugfix

**Careful points**
The `config dist.json` has been updated and two config field has been changed:
-  The Mongo DB require now the full connection URL instead of many differents fields.
- When SSL is set to `true`, it's possible to specify a `REDIRECT_PORT` field for listen http on this port and redirect on the main port.

### CODE REVIEW

File: src/config/config dist.json [hunk 1]
```diff
@@ -1,21 +1,16 @@
...
-        "DB_HOST": "db",
-        "DB_USERNAME": "root",
-        "DB_PASSWORD": "root",
-        "DB_DATABASE": "codewe",
-        "DB_PORT": "3306"
-    },
+    "REDIRECT_PORT": null,
     "METRICS": false,
     "METRICS_PORT": 8000
 }
```
Comment (TheoTechnicguy) [inline src/config/config dist.json:16]: ⚠️ **THIS IS A POTENTIALLY BREAKING CHANGE!** ⚠️ 
--------------------------------------------------------- 
What will break here? What and how will we change these setting in the production site `config.json`?

File: src/db/MongoDB.js [hunk 2]
```diff
@@ -30,29 +34,29 @@ class MongoDB {
...
             let results = (await this.documentsCollection.insertOne(doc));
             const documentLink = utils.uuid(results.insertedId.toString());
-            this.documentsCollection.updateOne({_id: results.insertedId}, {$set: {documentLink: documentLink}})
+            const linkView = utils.uuid(documentLink);
+            this.documentsCollection.updateOne({_id: results.insertedId}, {$set: {documentLink: documentLink, linkView: linkView}});
             return documentLink;
         } catch (err) {
             if (configs.DEBUG) {
                 console.error('Error when creating a new document');
             }
```
Comment (TheoTechnicguy) [inline src/db/MongoDB.js:59]: For debug, shouldn't we print out more information?

File: src/db/MongoDB.js [hunk 3]
```diff
@@ -64,94 +68,171 @@ class MongoDB {
...
 
 function getDB () {
-    db = new MongoDB(
-        configs.DB_CONFIG.DB_USERNAME,
-        configs.DB_CONFIG.DB_PASSWORD,
-        configs.DB_CONFIG.DB_HOST,
-        configs.DB_CONFIG.DB_DATABASE,
-        configs.DB_CONFIG.DB_PORT
-    );
+    db = new MongoDB(configs.DB_URL);
```
Comment (TheoTechnicguy) [inline src/db/MongoDB.js:235]: ⚠️ This is (also) a potentially breaking change ⚠️ 
--------------------------------------------------
How do we change the config in production?

File: src/server.js [hunk 4]
```diff
@@ -14,8 +14,9 @@ const host = configs.HOST;
 const port = configs.PORT;
 const DEBUG = configs.DEBUG;
 const ssl = configs.SSL;
+const metrics = configs.METRICS;
 const sslKeyPath = configs.KEY_FILE_SSL;
-const sslCertPath = configs.CERT_FILE_SSL; 
+const sslCertPath = configs.CERT_FILE_SSL;
```
Comment (TheoTechnicguy) [inline src/server.js:19]: Again, what will break here?

**Review by TheoTechnicguy — COMMENTED** [review]
Please let us note what we need to change in `config.json` for a smooth transition.

**Comment by BhasherBEL** [comment]
Description has been updated based on your suggestions.

File: src/db/MongoDB.js (hunk 3 above)
Comment (TheoTechnicguy) [reply src/db/MongoDB.js:235]: New description resolves this

File: src/server.js (hunk 4 above)
Comment (TheoTechnicguy) [reply src/server.js:19]: New description resolves this

**Review by TheoTechnicguy — APPROVED** [review]

**Review by allEyezOnCode — APPROVED** [review]

**Pull request merged**